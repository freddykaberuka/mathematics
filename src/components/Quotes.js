import React, { useState, useEffect } from 'react';

const Quotes = () => {
  const [quote, setQuote] = useState('');
  const [isLoading, setIsLoading] = useState(true);
  const [quoteError, setQuoteError] = useState('');

  useEffect(() => {
    const fetchQuotes = async () => {
      try {
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=computers',
          {
            headers: {
              'X-Api-Key': 'QpSPqtCezVld+xnq6vgFXw==wZwkCal20Yi17pOx',
            },
          },
        );
        const data = await response.json();
        setQuote(data[0].quote);
        setIsLoading(false);
      } catch (error) {
        setQuoteError('Error occurred fetching quotes!');
        setIsLoading(false);
      }
    };

    setTimeout(() => {
      fetchQuotes();
    }, 1000);
  }, []);

  return (
    <div>
      <div className="quotes">
        <h2>Computer Quotes</h2>
        {isLoading && <div>Loading...</div>}
        {quoteError && (
          <div>
            Error:
            {' '}
            {quoteError}
          </div>
        )}
        {!isLoading && !quoteError && (
          <div>
            <blockquote>
              `&quot;`
              {quote}
              `&quot;`
            </blockquote>
          </div>
        )}
      </div>
    </div>
  );
};

export default Quotes;
