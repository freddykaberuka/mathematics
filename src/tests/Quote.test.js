import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Quotes from '../pages/Quotes';

describe('Test Quotes rendering', () => {
  it('should render quotes correctly', () => {
    const quote = renderer.create(<Router><Quotes /></Router>).toJSON();
    expect(quote).toMatchSnapshot();
  });
});
