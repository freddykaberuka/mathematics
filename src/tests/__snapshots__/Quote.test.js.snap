// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Quotes rendering should render quotes correctly 1`] = `
<div>
  <div
    className="quotes"
  >
    <h2>
      Computer Quotes
    </h2>
    <div>
      Loading...
    </div>
  </div>
</div>
`;
